tools/serverpod_cli/lib/src/analyzer/dart/endpoint_analyzers/endpoint_class_analyzer.dart

```diff
@@ -40,11 +41,25 @@ abstract class EndpointClassAnalyzer {
     return true;
   }
 
-  /// Validates the [ClassElement] and returns a list of errors.
+  /// Validates the [ClassElement] and [endpointDefs] and returns a list of errors.
   static List<SourceSpanSeverityException> validate(
-    ClassElement classElement,
-  ) {
-    return [];
+      ClassElement? classElement, List<EndpointDefinition> endpointDefs) {
+    List<SourceSpanSeverityException> errors = [];
+
+    String? duplicateClassName = _checkForDuplicateClassNames(endpointDefs);
+    if (duplicateClassName != null) {
+      errors.add(
+        SourceSpanSeverityException(
+          'Endpoint analysis skipped due to duplicate class names. '
+          'Please rename your classes to make them unique. '
+          'className: $duplicateClassName',
+          null,
+          severity: SourceSpanSeverity.error,
+        ),
+      );
```

Comment:
i cant find a way to access the class element span, can you let me know how to do so, and couldnt find it anywhere in code

Reply:
It should be accessible on `classElement.span`. There is a method on the `ClassElement` class called `span` that gives back the `SourceSpan`.

Reply:
Sadly I can't check for multiple days, but I kept searching for it and couldn't find it

Reply:
It is an extension that needs to be imported: https://github.com/serverpod/serverpod/blob/main/tools/serverpod_cli/lib/src/analyzer/dart/element_extensions.dart

My IDEA handled the import for me automatically so I can understand that you missed it.

Reply:
It should, but for some reason serverpod repo takes roughly 18 gb from analysis server  , so I understand why it may failed to do the import for me.

I away from my laptop for 3 days, so I will make the pr a draft and convert it when I update to make it clear when it finished

Reply:
Awesome, looking forward to you completing the PR. This is an important contribution for the framework 🚀